packages/wonder-blocks-floating/src/components/floating.tsx

```diff
@@ -1,5 +1,208 @@
 import * as React from "react";
+import {
+    useFloating,
+    autoUpdate,
+    offset,
+    flip,
+    hide,
+    shift,
+    arrow,
+    Placement,
+} from "@floating-ui/react";
+import {StyleSheet} from "aphrodite";
+import {
+    border,
+    boxShadow,
+    semanticColor,
+} from "@khanacademy/wonder-blocks-tokens";
+import {addStyle} from "@khanacademy/wonder-blocks-core";
+import {ARROW_SIZE_INLINE} from "../util/constants";
+import {Arrow} from "./floating-arrow";
 
-export default function Floating() {
-    return <div>Floating</div>;
+const StyledDiv = addStyle("div");
+
+type FloatingProps = {
+    /**
+     * The reference (or anchored) element that is used to calculate the
+     * position of the floating element.
+     */
+    children: React.ReactElement;
+    /**
+     * The content to display in the floating element.
+     */
+    content: React.ReactNode;
+
+    /**
+     * The padding to apply between the floating element and its boundary.
+     * @default 8
+     */
+    boundaryPadding?: number;
+
+    /**
+     * The placement of the floating element relative to the reference element.
+     * @default "top"
+     * @see https://floating-ui.com/docs/useFloating#placement
+     */
+    placement?: Placement;
+
+    /**
+     * The strategy to use for positioning the floating element.
+     * @default "absolute"
+     * @see https://floating-ui.com/docs/useFloating#strategy
+     */
+    strategy?: "fixed" | "absolute";
+
+    /**
+     * Whether the floating element is open.
+     * @default false
+     */
+    open: boolean;
+    /**
+     * Callback for when the floating element is opened or closed.
+     */
+    onOpenChange?: (open: boolean) => void;
+
+    /**
+     * The test ID to use for the floating element.
+     */
+    testId?: string;
+
+    // ----- Middleware specific props -----
+    /**
+     * The flip strategy to use.
```

Comment:
Since the `flip` prop is a boolean, what do you think about re-phrasing the prop description since consumers don't choose a specific flip strategy! Something like "whether the floating element is flipped" or something!